libs/crowdfunding/editor/src/lib/create-crowdfunding-form.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  CreateCrowdfundingGoalInput,
+  CreateCrowdfundingInput,
+  getApiClientV2,
+  useCreateCrowdfundingMutation,
+  CreateCrowdfundingMutation
+} from '@wepublish/editor/api-v2'
+import {useMemo, useState} from 'react'
+import {useTranslation} from 'react-i18next'
+import {useNavigate} from 'react-router-dom'
+import {CrowdfundingForm} from './crowdfunding-form'
+import {SingleViewTitle} from '@wepublish/ui/editor'
+import {Form, Message, Schema, toaster} from 'rsuite'
+import {ApolloError} from '@apollo/client'
+
+const showError = (error: ApolloError): void => {
+  toaster.push(
+    <Message type="error" showIcon closable duration={3000}>
+      {error.message}
+    </Message>
+  )
+}
+
+export const CreateCrowdfundingForm = () => {
+  const navigate = useNavigate()
+  const {t} = useTranslation()
+  const client = useMemo(() => getApiClientV2(), [])
+
+  const closePath = '/crowdfundings'
+
+  const [crowdfunding, setCrowdfunding] = useState({} as CreateCrowdfundingInput)
```

Comment:
The way `setCrowdfunding` is currently used *can* lead to bugs as setStates can be batched and thus `crowdfunding` be outdated.
To make this easier we can use `useReducer` here instead, so instead of doing:

```
setCrowdfunding({
      ...crowdfunding,
      goals: [...(crowdfunding.goals || []), goal]
    })
```

one would do 

```
setCrowdfunding({
      goals: [...(crowdfunding.goals || []), goal]
    })
```

The useReducer would look like this:


```suggestion
  const [crowdfunding, setCrowdfunding] = useReducer((state: CreateCrowdfundingInput, action: Partial<CreateCrowdfundingInput>) => ({...state, ...action}), {})
```